src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -387,7 +389,12 @@ internal static void WriteEscapedJsonString(TextWriter writer, string inputStrin
                 for (; currentIndex < inputStringLength; currentIndex++)
                 {
                     char c = inputString[currentIndex];
-                    string escapedString = JsonValueUtils.SpecialCharToEscapedStringMap[c];
+                    string escapedString = null;
+
+                    if (stringEscapeOption == ODataStringEscapeOption.EscapeNonAscii || c < 0x7F)

```

Comment:
how about c==0x7F ??

Reply:
There is a misalignment for 0x7F.
When SpecialCharToEscapedStringMap is created (https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Json/JsonValueUtils.cs#L567-L569): 0x7F is considered non-special character. But, by using "<" here, 0x7F is considered special character.

Reply:
You can consider "0x7F" here as a threshold. A char is a special char or not is not identified from the value, it's from "specialCharToEscapedStringMap". 

owing that specialCharToEscapedStringMap[0x7f] == null, so it's not a special char.

Reply:
Is this one resolved?

Reply:
changed to use <=